### PR TITLE
Add `debug` to Gemfile

### PR DESCRIPTION
**Pull request by st0012**
This PR also bumps Sorbet for the updated `irb` rbi (PR: https://github.com/sorbet/sorbet/pull/6509)

### CODE REVIEW

**Review by KaanOzkan — APPROVED** [review]

File: Gemfile [hunk 1]
```diff
@@ -7,6 +7,7 @@ gemspec
 gem("minitest")
 gem("minitest-hooks")
 gem("minitest-reporters")
+gem("debug", require: false)
```
Comment (Morriar) [inline Gemfile:10]: Shouldn't it be in the `development` group?

Comment (st0012) [reply Gemfile:10]: It should at least be included in `test` as well?
I think it belongs to where `pry-byebug` sits, which is at the top-level atm.

**Review by vinistock — COMMENTED** [review]
Should we just remove `pry-byebug` and use only debug? Is there a reason we need to keep it?

**Review by vinistock — APPROVED** [review]
I vote for removing `pry-byebug`, but don't object keeping it if other people are using it

**Review by Morriar — APPROVED** [review]

**Pull request merged**